### ISSUE TITLE
'PyArray_Descr’ {aka ‘struct _PyArray_Descr’} has no member named ‘elsize’ when building using Forthon on GitHub runners

### ISSUE THREAD

**Issue by holm10** [dpgrote/Forthon#33]
Automated builds of UEDGE on GitHub runners using workflows started failing in last 2 months. The error seems to imply a Forthon issue:

```
       gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -DFORTHON_PKGNAME=\"uedge\" -I/tmp/pip-build-env-qqg6217r/overlay/lib/python3.9/site-packages/Forthon -I/tmp/pip-build-env-qqg6217r/overlay/lib/python3.9/site-packages/numpy/_core/include -I/opt/hostedtoolcache/Python/3.9.19/x64/include/python3.9 -c build/temp.linux-x86_64-cpython-39/compymodule.c -o build/temp.linux-x86_64-cpython-39/compymodule.o
      In file included from build/temp.linux-x86_64-cpython-39/compymodule.c:1:
      build/temp.linux-x86_64-cpython-39/Forthon.h: In function ‘Forthon_setarray’:
      build/temp.linux-x86_64-cpython-39/Forthon.h:860:58: error: ‘PyArray_Descr’ {aka ‘struct _PyArray_Descr’} has no member named ‘elsize’
        860 |           (((PyArrayObject_fields *)(farray->pya))->descr->elsize) = 0;
            |                                                          ^~
      build/temp.linux-x86_64-cpython-39/Forthon.h:863:58: error: ‘PyArray_Descr’ {aka ‘struct _PyArray_Descr’} has no member named ‘elsize’
        863 |           (((PyArrayObject_fields *)(farray->pya))->descr->elsize) = (int)PyArray_ITEMSIZE(ax);
            |                                                          ^~
      build/temp.linux-x86_64-cpython-39/Forthon.h:875:64: error: ‘PyArray_Descr’ {aka ‘struct _PyArray_Descr’} has no member named ‘elsize’
        875 |     if (d > -1) (((PyArrayObject_fields *)(farray->pya))->descr->elsize) = d;
            |                                                                ^~
      error: command '/usr/bin/gcc' failed with exit code 1
      make: *** [Makefile.Forthon:19: build/compydep] Error 1

```

Full logs available here: https://github.com/LLNL/UEDGE/actions/runs/9865600958
Fails on all build platforms. Happy to provide more information and help debugging. Will try to recreate locally.

**Comment by holm10** [comment]
Unable to recreate locally, and downgrading Forthon to 0.10.3 on the runner still reproduces the issue: possible that the error is not directly Forthon-related?

Edit: able to reproduce error when upgrading to Numpy 2.0 locally, likely related to the below referenced Numpy changes.

**Comment by holm10** [comment]
I think I found the culprit, which seems to be numpy, as described here:
https://github.com/numpy/numpy/commit/b3f9fc0ffed50ac437a2f09ecffeb6709a2487c8

**Comment by dpgrote** [comment]
This is now fixed in the latest version, 0.10.7.
